tests/blockchain/eth/contracts/integration/test_contracts_upgradability.py

```diff
@@ -0,0 +1,144 @@
+"""
```

Comment:
:rocket:  - This is a relief, good work.  I think this test can be emphasized in our CI workflow by running it in its own job.

Reply:
✔️